libs/sdk-backend-tiger/src/backend/workspaces/index.ts

```diff
@@ -67,12 +69,10 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
         return this;
     }
 
-    public withFilter(filter: {
-        description?: string;
-        earlyAccess?: string;
-        prefix?: string;
-    }): IWorkspacesQuery {
-        this.filter = filter;
+    public withFilter(filter: IWorkspacesQueryFilter): IWorkspacesQuery {
+        const { rootWorkspacesOnly, ...filterAttributes } = filter;
+        this.filter = filterAttributes;
+        this.rootWorkspacesOnly = !!rootWorkspacesOnly;
```

Comment:
This is matter of opinion but I would leave this as is and just sanitize it where you use it below to `!!this.filter.rootWorkspacesOnly && ...)

Reply:
@xMort the boolean cast would be unnecessary where I use it as `this.rootWorkspacesOnly && ...` since the params array is then filtered with Boolean (https://github.com/gooddata/gooddata-ui-sdk/pull/4944/files#diff-186fc90fac4c544f5629aa57478f32750dd70de37f7039c894694cf8175af815R135), so possibly we can get rid of the double-bang here, and just change the type of the property to `boolean | undefined`  here: https://github.com/gooddata/gooddata-ui-sdk/pull/4944/files#diff-186fc90fac4c544f5629aa57478f32750dd70de37f7039c894694cf8175af815R42 which I will have to do either way if we remote the double-bang here.

So do you prefer to change the prop type over the double-bang in the `withFilter` method? I can change it since I'll be removing the empty line [above](https://github.com/gooddata/gooddata-ui-sdk/pull/4944#discussion_r1579071537) anyway.

Reply:
I see, well I kept the filter property intact as it takes properties directly from the generated DeclarativeWorkspace interface, hence I wanted to keep the filter object consisting of only the direct workspace properties, but I can extend the filer prop type with this boolean prop of you prefer, no problem.

Also parentWorkspace, de facto a filter too, is also saved separately in the class hence I kept this one separately.

Reply:
@xMort modified, I put it in the filter: https://github.com/gooddata/gooddata-ui-sdk/compare/f7b8840b117549c7a35587b042844af7f9e88423..5d1467eb31a8e90966f799f8822d6c1677bf2ebb